lib/imageProcessor.js

```diff
@@ -47,11 +48,58 @@ ImageProcessor.prototype.analyze = function(path) {
 		})
 	})
 	.then(function(info) {
+		var dfd = when.defer()
+
+		try {
```

Comment:
a try catch around an asynchronous call doesn't do anything, and it's an anti-pattern in node.js